pkg/gardenlet/controller/managedseed/managedseed_controller.go

```diff
@@ -42,22 +63,34 @@ func (c *Controller) managedSeedAdd(obj interface{}) {
 		c.logger.Infof("Added ManagedSeed %s with delay %s to the queue", key, duration)
 		c.managedSeedQueue.AddAfter(key, duration)
 	}
+
 }
 
 func (c *Controller) managedSeedUpdate(_, newObj interface{}) {
 	managedSeed, ok := newObj.(*seedmanagementv1alpha1.ManagedSeed)
 	if !ok {
 		return
 	}
-	if managedSeed.Generation == managedSeed.Status.ObservedGeneration {
-		return
-	}
+
 	key, err := cache.MetaNamespaceKeyFunc(newObj)
 	if err != nil {
 		return
 	}
 
-	c.managedSeedQueue.Add(key)
+	jitterUpdates := (c.config.Controllers.ManagedSeed.JitterUpdates != nil) && *(c.config.Controllers.ManagedSeed.JitterUpdates)
+	if managedSeed.Generation == managedSeed.Status.ObservedGeneration {
+		return
+	} else {
+		if jitterUpdates {
+			duration := utils.RandomDurationWithMetaDuration(c.config.Controllers.ManagedSeed.SyncJitterPeriod)
+			c.logger.Infof("Added ManagedSeed %s with delay %s to the queue", key, duration)
+			c.managedSeedQueue.AddAfter(key, duration)
+		} else {
+			c.logger.Debugf("Added ManagedSeed %s without delay to the queue", key)
+			c.managedSeedQueue.Add(key)
+		}
+	}
+
```

Comment:
Can't we just call `managedSeedAdd(newObj)` here?

Reply:
In the case: `managedSeed.Generation == managedSeed.Status.ObservedGeneration` on Add we want to enqueue with delay (case of Gardenlet restart). But on Update we want to return without enqueuing.

Reply:
I have called `managedSeedAdd(newObj)` when `managedSeed.Generation != managedSeed.Status.ObservedGeneration` in [commit](https://github.com/gardener/gardener/pull/5483/commits/de284d6ecaa02c1411dda2dee6accfbb949d97f5).